### PR TITLE
MES-1622: Analytics test termination

### PR DESCRIPTION
**Pull request by jameshighfield**
## Description and relevant Jira numbers

Add analytics event for test termination from the test report page.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/test-report/__tests__/test-report.analytics.effects.spec.ts [hunk 1]
```diff
@@ -347,4 +348,22 @@ describe('Test Report Analytics Effects', () => {
...
+    it('should call logEvent for the termination event', () => {
+      // ARRANGE
+      spyOn(analyticsProviderMock, 'logEvent').and.callThrough();
+      store$.dispatch(new journalActions.StartTest(123456));
+      // ACT
+      actions.next(new testReportActions.TerminateTestFromTestReport());
+      // ASSERT
+      effects.testTermination$.subscribe((result) => {
+        expect(analyticsProviderMock.logEvent).toHaveBeenCalledWith(
+          AnalyticsEventCategories.TERMINATION,
```
Comment (Cavalier07) [inline src/pages/test-report/__tests__/test-report.analytics.effects.spec.ts]: you need a callback in this else it just skips the async stuff

File: src/pages/test-report/test-report.effects.ts [hunk 2]
```diff
@@ -138,4 +138,23 @@ export class TestReportEffects {
     delay(1000), // Added a 1 second delay to allow other action to complete/effects to fire
     map(() => new testsActions.PersistTests()),
   );
+
+  @Effect()
+  terminateTestFromTestReport$ = this.actions$.pipe(
+    ofType(
+      testReportActions.TERMINATE_TEST_FROM_TEST_REPORT,
+    ),
+    withLatestFrom(
```
Comment (johnmorleybjss) [inline src/pages/test-report/test-report.effects.ts]: Why select the current test if we don't use it for anything?

File: src/pages/test-report/__tests__/test-report.analytics.effects.spec.ts (hunk 1 above)
Comment (jameshighfield) [reply src/pages/test-report/__tests__/test-report.analytics.effects.spec.ts]: Good spot. Fixed.

File: src/pages/test-report/test-report.effects.ts (hunk 2 above)
Comment (jameshighfield) [reply src/pages/test-report/test-report.effects.ts]: Good find, have updated here: https://github.com/dvsa/mes-mobile-app/pull/564/files#diff-4ccf0d70ae9ceda46a4c611247cd9889R147

**Pull request merged**